### DISCUSSION TITLE
how to handle formData ?

### DISCUSSION THREAD

**Discussion by selcukguler0** [pocketbase/pocketbase#629] [Q&A]
hi,

i want to save image to store image. How can i handle formData coming from fetch?

im using multiparty library for parse file and getting this object 
```
[
  {
    fieldName: 'files',
    originalFilename: '456a49bd-ebc0-42f6-806c-8e43eb098476.JPG',
    path: '/var/folders/ty/v72_h2wd3xqffrhhztcfjs7r0000gn/T/Z7HR6TwYPihZc_jxlUyCZH93.JPG',
    headers: {
      'content-disposition': 'form-data; name="files"; filename="456a49bd-ebc0-42f6-806c-8e43eb098476.JPG"',
      'content-type': 'image/jpeg'
    },
    size: 115346
  }
]
```

nextjs api

```
import PocketBase from 'pocketbase';
import multiparty from 'multiparty';

const client = new PocketBase("http://11111:8090");

export const config = {
	api: {
		bodyParser: false,
	},
};

export default async function handler(req, res) {
	var form = new multiparty.Form();
	form.parse(req, async function (err, fields, files) {
		console.log(fields);
		console.log(files.files);
		const client = new PocketBase("http://11111:8090");
		const authData = await client.admins.authViaEmail('xxxxx@gmail.com', 'xxxxxxx');
		client.records.create("socket_machine", files)
			.then(async function (record) {
				res.status(200).json(record)

			}).catch(function (error) {
				res.status(500).json(error)
			});
	});
	
  
}
```

**Comment by ganigeorgiev** [comment]
I'm not familiar with `multiparty`, but in order to be able to submit `multipart/form-data` requests to the PocketBase API via the SDK, you'll need a proper [FormData object](https://developer.mozilla.org/en-US/docs/Web/API/FormData).

I can't find in the `multiparty` package documentation a helper for that, but you could try constructing it on your own based on the `fields` and `files` that you get in their `parse` callback:

```js
const data = new FormData();

// regular field
data.append("regularFieldName", "test"); // you may get the value from fields

// file field
// note: if the `files` items in your example are already Blob or File, manually reading each file may not be needed
data.append("fileUploadFieldName", new Blob([fs.readFileSync("./test.png")]), "test.png");

await client.records.create("socket_machine", data);
```

**Reply by selcukguler0** [reply]
> I'm not familiar with `multiparty`, but in order to be able to submit `multipart/form-data` requests to the PocketBase API via the SDK, you'll need a proper [FormData object](https://developer.mozilla.org/en-US/docs/Web/API/FormData).
> 
> I can't find in the `multiparty` package documentation a helper for that, but you could try constructing it on your own based on the `fields` and `files` that you get in their `parse` callback:
> 
> ```js
> const data = new FormData();
> 
> // regular field
> data.append("regularFieldName", "test"); // you may get the value from fields
> 
> // file field
> // note: if the `files` items in your example are already Blob or File, manually reading each file may not be needed
> data.append("fileUploadFieldName", new Blob([fs.readFileSync("./test.png")]), "test.png");
> 
> await client.records.create("socket_machine", data);
> ```

thank you for answer. 

Im using vercel for host and file writing not avilable. 
In my senerio user uploading photo. And im tring to send fomrData to backend with fetching api.
So after sending daha if i try to get formData like `req.body.formdata` its return undefined.

**Reply by ganigeorgiev** [reply]
> Im using vercel for host and file writing not avilable.

I'm not sure if Vercel disallow access to `fs`, but you don't actually need to write to the disk when doing `fs.readFileSync`. The idea is to pass the `file.path` to the fs.readFileSync for each file in order to generate a Blob (in your example the path is `/var/folders/ty/v72_h2wd3xqffrhhztcfjs7r0000gn/T/Z7HR6TwYPihZc_jxlUyCZH93.JPG`).


> So after sending daha if i try to get formData like req.body.formdata its return undefined.

You need to construct the FormData object manually. I doubt that  you can access the parsed multipart request body directly (otherwise you wouldn't need something like `multiparty`).
